firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt

```diff
@@ -46,18 +44,24 @@ internal constructor(
       blockingDispatcher,
       backgroundDispatcher,
       firebaseInstallations,
-      applicationInfo
+      applicationInfo,
     )
   private val sessionGenerator = SessionGenerator(collectEvents = shouldCollectEvents())
   private val eventGDTLogger = EventGDTLogger(transportFactoryProvider)
   private val sessionCoordinator = SessionCoordinator(firebaseInstallations, eventGDTLogger)
   private val timeProvider: TimeProvider = Time()
-  private val sessionStartScope = CoroutineScope(backgroundDispatcher)
 
   init {
     sessionSettings.updateSettings()
     val sessionInitiator =
-      SessionInitiator(timeProvider, this::initiateSessionStart, sessionSettings)
+      SessionInitiator(
+        timeProvider,
+        backgroundDispatcher,
+        object : SessionStartListener {
```

Comment:
This is so we don't have to make `onSessionStart` public.

Reply:
Can we add a comment for that?